C++/Limit Switch/src/main/cpp/Robot.cpp

```diff
@@ -39,17 +53,36 @@ class Robot : public frc::TimedRobot {
      * 
      * The isLimitSwitchEnabled() method can be used to check if the limit switch is enabled
      */
-    m_forwardLimit.EnableLimitSwitch(false);
-    m_reverseLimit.EnableLimitSwitch(false);
-    frc::SmartDashboard::PutBoolean("Forward Limit Enabled", m_forwardLimit.IsLimitSwitchEnabled());
-    frc::SmartDashboard::PutBoolean("Reverse Limit Enabled", m_reverseLimit.IsLimitSwitchEnabled());
+    maxConfig.limitSwitch
+      .ForwardLimitSwitchEnabled(false)
+      .ReverseLimitSwitchEnabled(false);
+
+    /**
+     * The ResetMode::kResetSafeParameters constant can be used to reset
+     * the configuration parameters in the SPARK MAX to their factory
+     * default state. If PersistMode::kNoPersistParameters is passed,
+     * these parameters will not persist between power cycles.
+     */
+    m_motor.Configure(maxConfig,
+      rev::spark::SparkMax::ResetMode::kResetSafeParameters,
+      rev::spark::SparkMax::PersistMode::kNoPersistParameters);
+
+    frc::SmartDashboard::PutBoolean("Forward Limit Enabled", m_motor.configAccessor.limitSwitch.GetForwardLimitSwitchEnabled());
+    frc::SmartDashboard::PutBoolean("Reverse Limit Enabled", m_motor.configAccessor.limitSwitch.GetReverseLimitSwitchEnabled());
   }
   void TeleopPeriodic() {
     m_motor.Set(m_stick.GetY());
 
+    rev::spark::SparkMaxConfig maxConfig;
+
     // enable/disable limit switches based on value read from SmartDashboard
-    m_forwardLimit.EnableLimitSwitch(frc::SmartDashboard::GetBoolean("Forward Limit Enabled", false));
-    m_reverseLimit.EnableLimitSwitch(frc::SmartDashboard::GetBoolean("Reverse Limit Enabled", false));
+    maxConfig.limitSwitch
+      .ForwardLimitSwitchEnabled(frc::SmartDashboard::GetBoolean("Forward Limit Enabled", false))
+      .ReverseLimitSwitchEnabled(frc::SmartDashboard::GetBoolean("Reverse Limit Enabled", false));
+
+    m_motor.Configure(maxConfig,
+      rev::spark::SparkMax::ResetMode::kNoResetSafeParameters,
+      rev::spark::SparkMax::PersistMode::kNoPersistParameters);
```

Comment:
We should remove this. It's going to constantly set the parameter and constantly block the program without checking if the dashboard value changes. If they want to enable it, they can do it in robotInit

Reply:
(this was a flaw in the original example)